Integrations/CounterCraft/CounterCraft.py

```diff
@@ -0,0 +1,873 @@
+
+""" COMMANDS """
+
+
+def test_module_command():
+
+    http_request("GET", "/campaigns")
+    demisto.results("ok")
+    sys.exit(0)
+
+
+def list_dsns_command():
+    """
+    Retrieve all deception support nodes
+    """
+
+    res = http_request("GET", "/deception_support_nodes")
+
+    new_dsns = []
+    for o in res["data"]:
+        new_dsns.append(
+            {
+                new_key: o[old_key] if old_key in o else None
+                for old_key, new_key in DSN_FIELDS.items()
+            }
+        )
+    context = createContext(new_dsns, removeNull=True)
+
+    return_results(
+        "Deception Support Node",
+        res["data"],
+        new_dsns,
+        {"CounterCraft.DSN(val.ID && val.ID === obj.ID)": context},
+        headers=["ID", "Name", "Description", "Hostname", "Port"],
+    )
+
+
+def list_providers_command():
+    """
+    Retrieve all providers
+    """
+
+    res = http_request("GET", "/providers")
+
+    new_providers = []
+    for o in res["data"]:
+        new_providers.append(
+            {
+                new_key: o[old_key] if old_key in o else None
+                for old_key, new_key in PROVIDER_FIELDS.items()
+            }
+        )
+    context = createContext(new_providers, removeNull=True)
+
+    return_results(
+        "Providers",
+        res["data"],
+        new_providers,
+        {"CounterCraft.Provider(val.ID && val.ID === obj.ID)": context},
+        headers=["ID", "Name", "Description", "TypeCode", "StatusCode"],
+    )
+
+
+def list_campaigns_command():
+    """
+    Retrieve all campaigns
+    """
+
+    name = demisto.args().get("name")
+
+    if name is not None:
+        criteria = {"criteria": "name:" + name}
+    else:
+        criteria = {}
+
+    res = http_request("GET", "/campaigns", params=criteria)
+
+    new_campaigns = []
+    for o in res["data"]:
+        new_campaigns.append(
+            {
+                new_key: o[old_key] if old_key in o else None
+                for old_key, new_key in CAMPAIGN_FIELDS.items()
+            }
+        )
+    context = createContext(new_campaigns, removeNull=True)
+
+    return_results(
+        "Campaigns",
+        res["data"],
+        new_campaigns,
+        {"CounterCraft.Campaign(val.ID && val.ID === obj.ID)": context},
+        headers=["ID", "Name", "Description", "StatusCode"],
+    )
+
+
+def list_hosts_command():
+    """
+    Retrieve all hosts
+    """
+
+    campaign_id = demisto.args().get("campaign_id")
+
+    if campaign_id is not None:
+        criteria = {"criteria": "campaigns.id:" + campaign_id}
+    else:
+        criteria = {}
+
+    res = http_request("GET", "/hosts", params=criteria)
+
+    new_hosts = []
+    for o in res["data"]:
+        new_hosts.append(
+            {
+                new_key: o[old_key] if old_key in o else None
+                for old_key, new_key in HOST_FIELDS.items()
+            }
+        )
+    context = createContext(new_hosts, removeNull=True)
+
+    return_results(
```

Comment:
in this case, it would be great also to output to [host context standard](https://demisto.pan.dev/docs/context-standards/#host)

Reply:
On the other hand, these hosts can also output to the host context standard.  Can we add them to the next release? :)

Reply:
Done!